apps/lib/apps.c

```diff
@@ -2379,56 +2392,57 @@ int raw_write_stdout(const void *buf, int siz)
  * don't free up other resources on the error path because they know it will
  * be handled when the process exits.
  */
-static BIO *check_bio(BIO *bp, const char *what, int mode)
+static BIO *check_bio(BIO *bp, const char *what, char mode, int bflags)
 {
     if (bp == NULL)
-        app_bail_out("%s: Could not open %s in mode 0x%x\n",
-                     opt_getprog(), what, mode);
+        app_bail_out("%s: Could not open %s for %s (BIO flags 0x%x)\n",
+                     opt_getprog(), what, modeverb(mode), bflags);
```

Comment:
This is definitely an improvement along the lines I had asked for some two weeks back.